samples/apps/copilot-chat-app/webapp/tests/chat.test.ts

```diff
@@ -1,51 +1,50 @@
-import { expect, test } from '@playwright/test';
+/* eslint-disable testing-library/prefer-screen-queries */
+import { test } from '@playwright/test';
+import * as util from './utils'
+import * as simpletests from './testsBasic'
+import * as plannertests from './testsPlanner'
+import { multiUserTest } from './testsMultiuser'
 
-/*
-This test is a simple end-to-end test that verifies the app can be
-loaded, and signed in. The user can send a message to the bot, and expect
-a response.
-*/
-test('get response from bot', async ({ page }) => {
-    // Make sure the server is running.
-    await page.goto('https://localhost:40443/healthz');
-    expect(page.getByText('Healthy')).toBeDefined();
+test.describe('Copilot Chat App Test Suite', () => {
+    // Note: A new chat session is opened for each test so that 
+    // the chat history is not polluted and the LLM is not confused.
 
-    await page.goto('/');
-    // Expect the page to contain a "Login" button.
-    await page.getByRole('button').click();
-    // Clicking the login button should redirect to the login page.
-    await expect(page).toHaveURL(new RegExp('^' + process.env.REACT_APP_AAD_AUTHORITY));
-    // Login with the test user.
-    await page.getByPlaceholder('Email, phone, or Skype').click();
-    await page.getByPlaceholder('Email, phone, or Skype').fill(process.env.REACT_APP_TEST_USER_ACCOUNT as string);
-    await page.getByRole('button', { name: 'Next' }).click();
-    await page.getByPlaceholder('Password').click();
-    await page.getByPlaceholder('Password').fill(process.env.REACT_APP_TEST_USER_PASSWORD as string);
-    await page.getByRole('button', { name: 'Sign in' }).click();
+    test.describe('A, runs in parallel with B', () => {
+        test.describe.configure({ mode: 'parallel' });
+        // Server Tests
+        test('Server Health', async ({ page }) => { await simpletests.serverHealth(page) });
+        // Basic Operations
+        test('Basic Bot Responses', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await simpletests.basicBotResponses(page) });
```

Comment:
why not the MS Graph plugin too?

Reply:
MS Graph connects to more external Microsoft Apps, like Calendar and email and todo, and to properly validate the tests we would need to do more work to access the APIs better. It was a lot of work and out of scope.
Plus the Core Kernel does a lot of testing for the MS Graph Skill already;